scanner/config.go

```diff
@@ -12,6 +12,36 @@ import (
 	"github.com/bitrise-io/go-utils/sliceutil"
 )
 
+const otherProjectType = "other"
+
+type scanResultStatus string
+
+const (
+	// in case DetectPlatform() returned error, or false
+	scanResultNotDetected scanResultStatus = "scanResultNotDetected"
+	// in case DetectPlatform() returned true, but Options() or Config() returned an error
+	scanResultDetectedWithErrors scanResultStatus = "scanResultDetectedWithErrors"
+	// in case DetectPlatform() returned true, Options() and Config() returned no error
+	scanResultDetected scanResultStatus = "scanResultDetected"
+)
+
+type scannerOutput struct {
+	scanResult scanResultStatus
+
+	// can always be set
+	// warnings returned by DetectPlatform(), Options()
+	warnings models.Warnings
+
+	// set if scanResultStatus is scanResultDetectedWithErrors
+	// errors returned by Config()
+	errors models.Errors
+
+	// set if scanResultStatus is scanResultDetected
+	optionModel      models.OptionNode
```

Comment:
since in this pr we are doing refactroing as well, may i would call it: `options`
an the `configMap` could be `configs`

Reply:
done